sdk/core/Azure.Core/src/Diagnostics/AzureEventSourceListener.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using Azure.Core.Shared;
+
+namespace Azure.Core.Diagnostics
+{
+    public class AzureEventSourceListener: EventListener
+    {
+        private readonly List<EventSource> _eventSources = new List<EventSource>();
+
+        private readonly Action<EventWrittenEventArgs, string> _log;
+        private readonly EventLevel _level;
+
+        public AzureEventSourceListener(Action<EventWrittenEventArgs, string> log, EventLevel level)
```

Comment:
Should we create our own type that's `EventWrittenEventArgs + string` with a potentially better name?